src/vm.rb

```diff
@@ -61,6 +61,61 @@ def destroy
 
     private
 
+    CERTS_DIR = '/var/lib/metalware/certs'
+
+    def libvirt_host
+      node.config.libvirt_host
+    end
+
+    def libvirt
+      Libvirt.open("qemu://#{libvirt_host}/system")
+    end
+
+    def certificate?
+      File.exist?("#{CERTS_DIR}/#{libvirt_host}-key.pem")
+    end
+
+    def generate_certificate
+      generate_certificate_key
+      generate_certificate_info
+      generate_server_certificates
+      puts certificate_generation_message
+      exit
+    end
+
+    def generate_certificate_key
+      SystemCommand.run("certtool --generate-privkey > #{CERTS_DIR}/#{libvirt_host}-key.pem")
+    end
+
+    def generate_certificate_info
+      File.open("#{CERTS_DIR}/#{libvirt_host}.info", 'w') do |f|
```

Comment:
As above, this is duplicated here and in `generate_server_certificates` so it would make sense to split out a `certificate_info_path` method for it, and to also use `File.join` while doing that.